src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs

```diff
@@ -267,16 +272,25 @@ public override async ValueTask<FormulaValue> Visit(CallNode node, EvalVisitorCo
             {                
                 return await asyncConnectorTexlFunction.InvokeAsync(args, _services, _cancellationToken).ConfigureAwait(false);
             }
-            else if (func is UserDefinedTexlFunction udtf)
-            {
-                // $$$ Should add _runtimeConfig
-                result = await udtf.InvokeAsync(args, _cancellationToken, context.StackDepthCounter.Increment()).ConfigureAwait(false);
-            }
             else if (func is CustomTexlFunction customTexlFunc)
             {
                 // If custom function throws an exception, don't catch it - let it propagate up to the host.
                 result = await customTexlFunc.InvokeAsync(FunctionServices, args, _cancellationToken).ConfigureAwait(false);
             }
+            else if (func is UserDefinedFunction userDefinedFunc)
+            {
+                userDefinedFunc.BindBody(_symbolValues.SymbolTable, new Glue2DocumentBinderGlue(), bindingConfig: BindingConfig.Default);
+
+                _formulaValuesStack.Push(args);
+
+                (var irnode, _) = userDefinedFunc.GetIRTranslator();
+
+                this.CheckCancel();
+
+                result = await irnode.Accept(this, context).ConfigureAwait(false);
+
+                _formulaValuesStack.Pop();

```

Comment:
> _formulaValuesStack.Pop(); [](http://example.com/codeflow?start=15&length=27)

nit - I'd put this in a finally - just in case an exception is thrown.